python/GafferArnoldUI/ArnoldShaderUI.py

```diff
@@ -336,6 +345,15 @@ def __translateNodeMetadata( nodeEntry ) :
 				"ai:light", paramName, page
 			)
 
+		if (
+			nodeName in lightEditorShaders and
+			__aiMetadataGetStr( nodeEntry, paramName, "gaffer.plugType" ) != ""
+		) :
+			attributeName, sectionName = lightEditorShaders[nodeName]
+			GafferSceneUI.LightEditor.registerShaderParameter(
+				"ai:light", paramName, attributeName, sectionName
+			)
+
```

Comment:
Yes, that's a nice attribute to include for sure. I added that in https://github.com/GafferHQ/gaffer/pull/5895/commits/583ca5021bbb46f0375e10d556d89e4cd35c7a1a and https://github.com/GafferHQ/gaffer/pull/5895/commits/5f0036f3463d53d50c563e9bc9e9bbc90bb93102.

Adding "Select Affected Objects" seems nice too. I added that to keep parity with the Pass Editor (and you had already done the hard part). It was a pretty straightforward repurposing of the Pass Editor code so maybe it won't be too much trouble to unify the two. That's done in https://github.com/GafferHQ/gaffer/pull/5895/commits/5f0036f3463d53d50c563e9bc9e9bbc90bb93102

Reply:
Thanks! I think we'll just need to add "filteredLights" to the attribute registry over in [EditScopeAlgo](https://github.com/GafferHQ/gaffer/blob/dd6f30d5a01f42d57e13f057e16048b325a22d79/src/GafferScene/EditScopeAlgo.cpp#L85-L93), so this can still be edited in an EditScope when there's no upstream attribute.

There'd also be an argument for allowing this to be edited using the new SetExpressionPlugValue widget, but I think we can handle that in a separate PR as it'll require a little additional metadata and AttributeTweaks plumbing along the lines of what was done for OptionTweaks over here: https://github.com/GafferHQ/gaffer/pull/5871/commits/4bc8ef682998ba571e9b3c4bdf16a6bb85d10e5c

Reply:
Right, that explains why I needed to set the value in the unit test. I added that registration, and removed the manual value setting in the unit test to verify the registration worked. Those are in bdf665d8748817f1a12e98a456e83ac514160b11

Reply:
> There'd also be an argument for allowing this to be edited using the new SetExpressionPlugValueWidget

I looked at adding that too, but I realized that it may not be as simple. If you're editing the plug on the blocker itself - not a tweak or Edit Scope - the `SetExpressionPlugValueWidget` will only be somewhat useful, and might even be misleading. Since there is no input to a light blocker, the widget doesn't add anything useful by way of autocomplete or browsing existing sets.

It would be useful for modifying `filteredLights` in a tweak plug / Edit Scope though. Maybe we could manage a conditional where we decide which widget to use based on the source plug?